src/promptflow/promptflow/batch/_batch_inputs_processor.py

```diff
@@ -0,0 +1,296 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from pathlib import Path
+from typing import Any, Dict, List, Mapping, Optional
+
+from promptflow._core._errors import UnexpectedError
+from promptflow._utils.load_data import load_data
+from promptflow._utils.logger_utils import logger
+from promptflow._utils.multimedia_utils import resolve_multimedia_data_recursively
+from promptflow._utils.utils import resolve_dir_to_absolute
+from promptflow.batch._errors import InputMappingError
+from promptflow.contracts.flow import FlowInputDefinition
+from promptflow.executor.flow_executor import LINE_NUMBER_KEY
+
+
+class BatchInputsProcessor:
+    def __init__(
+        self,
+        working_dir: Path,
+        flow_inputs: Dict[str, FlowInputDefinition],
```

Comment:
This type should be Mapping instead of Dict? since we actually should not edit it.

Reply:
Updated.